lti_consumer/lti_xblock.py

```diff
@@ -314,6 +314,12 @@ class LtiConsumerXBlock(StudioEditableXBlockMixin, XBlock):
             "prior to doing the launch request."
         ),
     )
+    lti_1p3_redirect_uris = List(
+        display_name=_("Registered Redirect URIs"),
+        # TODO: Further clarify this help
+        help=_("Valid URLs the Tool may redirect you to. This may or may not be the same as the launch url."),
```

Comment:
I know you have a TODO on this, so perhaps you already have this in mind, but thought I'd mention it anyway. I think this help text is a little confusing as is. The redirect URIs represents URIs that the platform is allowed to redirect to on the tool's side, as specified by the tool during the preregistration phase. It's the platform that issues a redirect to one of these URIs, not the tool. The tool instructs the platform which of these URIs to use in the authentication request.

Regarding, "This may or may not be the same as the launch url.", is it your experience that tools don't explicitly specify redirect URIs and rely on the person setting up the integration to know that this is the launch URL otherwise? I haven't set up live LTI 1.3 tools yet - only the 1EdTech testing suite, which explicitly provides the redirect URIs. I just worry that course team authors or other users will just stick the launch URL in here.

Reply:
Correct, it was confusing which is why I left the TODO! I figured it would be something we needed to discuss. But yes your description is otherwise accurate.

> Is it your experience that tools don't explicitly specify redirect URIs and rely on the person setting up the integration to know that this is the launch URL otherwise?

No, the tool must always specify the allowed redirect uri's as that is part of the OIDC spec which is what LTI 1.3 security is modeled off. The point is more that the `target_link_uri` (what we call launch url) and the `redirect_uri` may not always be the same. As you stated the `redirect_uri` is where the token is POSTed to and must be registered on the platform, while the `target_link_uri` is where the tool would redirect you to after that step in order to display the final content.